src/it/java/com/google/checkstyle/test/chapter5naming/rule526parameternames/ParameterNameTest.java

```diff
@@ -43,36 +49,63 @@ protected String getPath(String fileName) throws IOException {
 
     @BeforeClass
     public static void setConfigurationBuilder() throws CheckstyleException {
-        checkConfig = getCheckConfig("ParameterName");
-        format = checkConfig.getAttribute("format");
+        final List<Configuration> configs = getCheckConfigs("ParameterName");
+
+        Assert.assertEquals(configs.size(), 2);
+
+        privConfig = configs.get(0);
+        Assert.assertEquals(privConfig.getAttribute("excludeScope"), "public");
+        privFormat = privConfig.getAttribute("format");
+
+        pubConfig = configs.get(1);
+        Assert.assertEquals(pubConfig.getAttribute("scope"), "public");
+        pubFormat = pubConfig.getAttribute("format");
+    }
+
+    @Test
+    public void privParameterNameTest() throws Exception {
+
+        final String[] expected = {
+            "8:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "$arg1", privFormat),
+            "9:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "ar$g2", privFormat),
+            "10:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "arg3$", privFormat),
+            "11:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "a_rg4", privFormat),
+            "12:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "_arg5", privFormat),
+            "13:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "arg6_", privFormat),
+            "14:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "aArg7", privFormat),
+            "15:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "aArg8", privFormat),
+            "16:21: " + getCheckMessage(privConfig.getMessages(), MSG_KEY, "aar_g", privFormat),
+        };
+
+        final String filePath = getPath("InputParameterNameSimple.java");
+
+        Integer[] warnList = getLinesWithWarn(filePath);
+        final List<Integer> selected = new ArrayList<>();
+        Collections.addAll(selected, warnList);
+        selected.removeIf(line -> line > 16);
```

Comment:
I agree, I will split it into two input files: one containing only private methods and the other contacting public methods.